mm/2s2h/Rando/ActorBehavior/ObjGrass.cpp

```diff
@@ -1061,29 +1066,46 @@ void Rando::ActorBehavior::InitObjGrassBehavior() {
         }
     });*/
 
+    COND_VB_SHOULD(VB_GRASS_STORE_RANDO_CHECK_ID, IS_RANDO, {
+        grassCarryHome = gZeroVec3f;
+        ObjGrassCarry* grassActor = va_arg(args, ObjGrassCarry*);
+
+        grassCarryHome = grassActor->actor.world.pos;
+        grassCarryHome.x = roundUpToTwoDecimalPlaces(grassCarryHome.x);
+        grassCarryHome.z = roundUpToTwoDecimalPlaces(grassCarryHome.z);
+
+        *should = false;
+    });
+
     COND_VB_SHOULD(VB_GRASS_DROP_COLLECTIBLE, IS_RANDO, {
         auto actorId = static_cast<ActorId>(va_arg(args, int32_t));
         ObjGrassElement* grassElemActor;
         EnKusa* kusaActor;
-        Vec3f pos;
+        ObjGrassCarry* grassCarryActor;
+        Vec3f pos = gZeroVec3f;
 
         if (actorId == ACTOR_OBJ_GRASS) {
             grassElemActor = va_arg(args, ObjGrassElement*);
             pos = grassElemActor->pos;
-        } else {
+        } else if (actorId == ACTOR_EN_KUSA) {
             kusaActor = va_arg(args, EnKusa*);
             pos = kusaActor->actor.home.pos;
+        } else if (actorId == ACTOR_OBJ_GRASS_CARRY) {
+            grassCarryActor = va_arg(args, ObjGrassCarry*);
+            pos = grassCarryActor->actor.world.pos;
```

Comment:
Instead of using/adding `VB_GRASS_STORE_RANDO_CHECK_ID` and relying on updating a static var, you should be able to get the original grass elem from carry actor and get its pos value like this:

```
pos = grassCarryActor->grassElem->pos;
```

Reply:
I glossed over the ObjGrassCarry having the grassElem in it... this coulda saved me ever going down the static var route to begin with... I just thought it was gone entirely..